acme/acme/client.py

```diff
@@ -1106,10 +1109,15 @@ def _add_nonce(self, response):
         else:
             raise errors.MissingNonce(response)
 
-    def _get_nonce(self, url):
+    def _get_nonce(self, url, acme_version):
         if not self._nonces:
             logger.debug('Requesting fresh nonce')
-            self._add_nonce(self.head(url))
+            if acme_version == 1:
+                response = self.head(url)
+            else:
+                # request a new nonce from the acme newNonce endpoint
+                response = self._check_response(self.head(self.new_nonce_url), content_type=None)
```

Comment:
I think this would throw a weird error if `new_nonce_url` is `None`. Can we check that case and raise an error that says what's wrong?